Muszilla/src/Controller/HomeController.php

```diff
@@ -0,0 +1,309 @@
+use App\model\User;
+use App\model\Music;
+use Exception;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Routing\Annotation\Route;
+
+/**
+ * Class to handle all the routes available in the Music web app. It handle the 
+ * routing and the operations involved with that particular route.
+ * 
+ * @var object $conn
+ *    Variable to store the mysqli database connection object.
+ * 
+ * @var object $music 
+ *    Variable to store the object of the Music class.
+ * 
+ * @var object $connection_obj 
+ *    Variable to store the object od the DbConnect class.
+ * 
+ * @var object $user 
+ *    Variable to store the object of the User class.
+ * 
+ * @var object $upload_obj
+ *    Variable to store the object of the uploadCon class.
+ */ 
+class HomeController extends AbstractController {
+
+  private object $conn = '';
+  private object $music = '';
+  private object $connection_obj = '';
+  private object $upload_obj = '';
+  private object $user = '';
+
+  /**
+   * Construction to initialize the objects of the classes used in the 
+   * HomeController.
+   */
+  public function __construct() {
+    $this->music = new Music();
+    $this->connection_obj = new DbConnect();
+    $this->conn = $this->connection_obj->connect();
+    $this->upload_obj = new uploadCon();
+    $this->user = new User();
+  }
+
+  #[Route('/', name: 'app_login')]
+  /**
+   * Root Route to handle the user request to the login page.
+   *
+   * @param Request $request
+   *    Stores the credentials entered by the user.
+   * 
+   * @return Response
+   *    If the user credentiaals are valid, it redirect the user to the Home 
+   *    page,
+   *    otherwise redirects the user to the login page. 
+   */
+  public function login(Request $request): Response {
+    session_start();
+    if(isset($_SESSION['email'])) {
+      $user = $this->user->getUserData($_SESSION['email']);
+      return $this->render('Muszilla/home.html.twig', ['user'=>$user]);
+    }
+    if($_SERVER['REQUEST_METHOD'] == 'POST') {
+      try {
+        $data = $request->request->all();
+        $img = $this->upload_obj->uploadImage($data);
+        $register = $this->user->register($data, $img);
+      }
+      catch (Exception $e){
+        $error = $e;
+        $user = $this->user->getUserData($_SESSION['email']);
+        return $this->render('Muszilla/error.html.twig', ['error' => $error, 'user' => $user]);
+      }
+      
+    }
+    return $this->render('Muszilla/login.html.twig');
+  }
+
+
+  #[Route('/', name: 'app_root')]
+  /**
+   * Root route to handle the condition when the user tries to go back to the 
+   * login page after logging in.
+   *
+   * @return Response
+   *    Returns the home page if the user is already logged in, otherwise 
+   *    redirects the user to the home page.
+   */
+  public function root(): Response {
```

Comment:
I think you can add this logic in above function!

Reply:
Yes, that can be Done. Did it just to make the home route look clean. Thanks, for suggesting.